jd-geek-source/src/app/services/api.services.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+@Injectable()
+
+export class ApiService {
+  constructor(private http: HttpClient) {}
+  contentTypeQuery = '?Content-type=%22application%2Fjson%22 HTTP/1.1';
+  headers = new HttpHeaders({'content-type': 'application/json'});
+
+  getTestData() {
+    var format = "JSON";
+    var apiKey = "ma6v7d7sxqmp6tbsssu7rcas"
```

Comment:
API keys should typically be stored as constants capable of being consumed throughout the app. Environment files are an ideal place to store keys.
https://angular.io/guide/build

Reply:
Yes.  I just wanted to see data come across.  I like to get the concept of pieces down and make sure I know how they work.